app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java

```diff
@@ -15,114 +19,142 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.util.EmptyStackException;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
+import java.util.Stack;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
 
 public class TodoTxtFilter {
```

Comment:
just to be clear - basically the new filtering complete replaces the previous todo filter stuff including filter saving by you right? So need to make sure we use new preference keys for the stuff and get rid of the existing ones

Reply:
The `loadSavedViews` function will delete all the saved filters / views if it encounters a format it can't parse. This means that

1. When this pr is deployed, all the existing filters will get deleted on first load of the filter dialog
2. If there is malformed json (not sure how this could happen), the filters would be reset to a clean (empty state) rather than the dialog closing / not working.